dmscripts/send_dos_opportunities_email.py

```diff
@@ -67,47 +95,122 @@ def get_html_content(briefs, number_of_days):
+                'digital-specialists': [brief4, brief5],
+                'digital-outcomes': [brief6, brief7]
+            }
+        }
+    """
+    briefs = get_live_briefs_between_two_dates(client, start_date, end_date)
+    briefs_by_fw_and_lot = {}
 
+    for brief in briefs:
+        brief_fw_slug = brief['frameworkSlug']
+        brief_lot_slug = brief['lotSlug']
+        if brief['frameworkSlug'] in briefs_by_fw_and_lot:
+            if brief_lot_slug in briefs_by_fw_and_lot[brief_fw_slug]:
+                briefs_by_fw_and_lot[brief_fw_slug][brief_lot_slug].append(brief)
+            else:
+                briefs_by_fw_and_lot[brief_fw_slug][brief_lot_slug] = [brief]
+        else:
+            briefs_by_fw_and_lot[brief_fw_slug] = {
+                brief_lot_slug: [brief]
+            }
+    return briefs_by_fw_and_lot
+
+
+def main(data_api_client, mailchimp_client, number_of_days, framework_override, list_id_override, lot_slug_override):
+    """
+    Default behaviour is to fetch any briefs (regardless of framework/lot) published in the last N days.
+
+    The script groups briefs by framework/lot, and for each combination, looks up the Mailchimp list ID, and creates &
+    sends a new campaign.
+    - If a framework_override is supplied, just find briefs for that framework
+    - If a lot_override is supplied, just find briefs for that lot
+    - If a list_id_override is supplied, only send emails to that list ID regardless of lot/framework
+
+    For most of the year, there should only be one framework iteration with live briefs. However during the transition
+    period between DOS framework iterations, we need to support both iterations.
+
+    For example, on the second day of DOS4, any DOS3 briefs created the first day (before the go-live switchover) will
+    still need to be emailed to DOS3 suppliers, as well as the usual DOS4 briefs sent to DOS4 suppliers.
+    """
     start_date = date.today() - timedelta(days=number_of_days)
     end_date = date.today() - timedelta(days=1)
 
-    live_briefs = get_live_briefs_between_two_dates(
-        data_api_client, lot_data["lot_slug"], start_date, end_date, framework_slug
-    )
-    if not live_briefs:
+    # Get all live briefs for each DOS framework on the lot (or a single DOS framework, if supplied as a script arg)
+    live_briefs_by_framework = get_live_briefs_by_framework_and_lot(data_api_client, start_date, end_date)
+
+    # If specific framework script arg supplied, ignore other frameworks
+    if framework_override:
+        if not live_briefs_by_framework.get(framework_override):
+            logger.info(
+                "No new briefs found for {} in the last {} day(s)".format(framework_override, number_of_days)
+            )
+            return True
+        live_briefs_by_framework = {framework_override: live_briefs_by_framework.get(framework_override)}
+
+    # If no briefs found, exit early
+    if not live_briefs_by_framework.keys():
```

Comment:
I think `not live_briefs_by_framework` would have the same value as `not live_briefs_by_framework.keys()`

```suggestion
    if not live_briefs_by_framework:
```